lib/active_merchant/billing/gateways/trust_commerce.rb

```diff
@@ -214,18 +218,24 @@ def credit(money, identification, options = {})
       # TrustCommerce to allow for reversal transactions before you can use this
       # method.
       #
+      # void() is also used to to cancel a capture (postauth), purchase (sale),
+      # or refund (credit) or a before it is sent for settlement.
+      #
       # NOTE: AMEX preauth's cannot be reversed. If you want to clear it more
       # quickly than the automatic expiration (7-10 days), you will have to
       # capture it and then immediately issue a credit for the same amount
       # which should clear the customers credit card with 48 hours according to
       # TC.
       def void(authorization, options = {})
+        transaction_id, original_action = split_authorization(authorization)
+        action = (VOIDABLE_ACTIONS - ['preauth']).include?(original_action) ? 'void' : 'reversal'
```

Comment:
Just want to make sure I understand what's going on here. We must use `'void'` for voidable actions except `'preauth'`, which should still use `'reversal`' instead? I'm guessing this was the nuance we're trying to update for. And thus, even though there's a difference when voiding, we needed to append the original action for all action listed in `VOIDABLE_ACTIONS` so we can do this sort of check? Thus the need to use a subset of `VOIDABLE_ACTIONS` to determine which action to actually use for this void request?

Since `['preauth']` is a constant value, and is the exception in the case of void vs. reversal, maybe it would make sense to move it into a named constant like `VOIDABLE_VIA_REVERSAL_ACTIONS` or something similar? (Would also avoid allocating a new array on every call to `void`).

Reply:
I don't care for the single element array with `preauth`, but opted for explicitness because there is only a single action that uses `reversal` and it is only used in a single place. 

Moving the value to a constant would only prevent a new array for `void` if we were using a singleton instance of the gateway. If we instantiate a new gateway adapter for each transaction, a by-product of moving to a constant is that we would create this new constant array for every adapter, not just for void calls.

Reply:
> If we instantiate a new gateway adapter for each transaction, a by-product of moving to a constant is that we would create this new constant array for every adapter...

I won't insist on the constant, but I will point out that the constant reference is not re-initialized on class instancing:

```
irb(main):001:0> class Foo; BAR = ['baz']; def bar; BAR.object_id; end; end
=> :bar
irb(main):002:0> f1 = Foo.new
=> #<Foo:0x00007fd9d424c448>
irb(main):003:0> f2 = Foo.new
=> #<Foo:0x00007fd9d4254e18>
irb(main):004:0> f1.bar == f2.bar
=> true
```

Reply:
I thought about that when I went for a walk. You are right on the constant reference initialization, I was mixing languages.